src/main/java/org/elasticsearch/common/lucene/search/function/ScoreFunction.java

```diff
@@ -19,18 +19,16 @@
 
 package org.elasticsearch.common.lucene.search.function;
 
-import org.apache.lucene.index.LeafReaderContext;
 import org.apache.lucene.search.Explanation;
+import org.elasticsearch.common.lucene.ReaderContextAware;
 
 /**
  *
  */
-public abstract class ScoreFunction {
+public abstract class ScoreFunction implements ReaderContextAware {
```

Comment:
Indeed, but I made ReaderContextAware.setNextReader throw an io exception, and some score functions happen to delegate to other ReaderContextAware instances, so I had to make its `setNextReader` method throw an io exception too. But then it had exactly the same signature as the one in ReaderContextAware, so I just made ScoreFunction implement ReaderContextAware. (hope it makes sense)